### PR TITLE
Add MachinePools to CAPI apiversion upgrade test

### PR DESCRIPTION
**Pull request by willie-yao**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: test/e2e/data/infrastructure-azure/v1.17.0/cluster-template-prow-machine-and-machine-pool.yaml [hunk 1]
```diff
@@ -0,0 +1,920 @@
...
+        nodeRegistration:
+          kubeletExtraArgs:
+            cloud-provider: external
+          name: '{{ ds.meta_data["local_hostname"] }}'
+      preKubeadmCommands: []
+---
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: MachineDeployment
+metadata:
+  name: ${CLUSTER_NAME}-md-win
```
Comment (nojnhuh) [inline test/e2e/data/infrastructure-azure/v1.17.0/cluster-template-prow-machine-and-machine-pool.yaml:279]: Is there a reason that the "latest" version template doesn't include any Windows nodes like the 1.16.1 and 1.17.0 templates do?

Comment (willie-yao) [reply test/e2e/data/infrastructure-azure/v1.17.0/cluster-template-prow-machine-and-machine-pool.yaml:279]: I'm not too sure, but this is the same pattern from the existing `cluster-template-prow.yaml`: https://github.com/willie-yao/cluster-api-provider-azure/blob/test-aks-upgrade-1/test/e2e/data/infrastructure-azure/v1.16.1/cluster-template-prow.yaml#L217

The latest version template for `cluster-template-prow.yaml` also doesn't have windows nodes. I'm not exactly sure why that's the case but I just kept it consistent here.

**Pull request merged**